pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -2066,8 +2071,21 @@ class Timedelta(_Timedelta):
                 other = int(other)
             if isinstance(other, cnp.floating):
                 other = float(other)
+
+            # If we have a non-nano timedelta and
+            # self._value < other, we would get 0 as a result,
+            # which is not correct.
+            # We need to try going to a higher resolution to fix this
+            curr_value = self._value
```

Comment:
I'm not the happiest with this code, since there's a possibility we might overflow (if other is very big), and there could be the possibility that a Tick is not representable in nanoseconds.

Is this a real possibility?
(not too familiar with tick code, so I can't tell mysel)

Reply:
this introduces (untested) changes in Timedelta division in addition to Tick.  I'm skeptical that this is the right place for a change, but if it is, it should be tested.